bindings/sundials4py/cvodes/cvodes_generated.hpp

```diff
@@ -0,0 +1,2035 @@
+      nb::arg("cvode_mem"), nb::arg("dgmax_lsetup"));
+
+m.def("CVodeSetInitStep", CVodeSetInitStep, nb::arg("cvode_mem"), nb::arg("hin"));
+
+m.def("CVodeSetLSetupFrequency", CVodeSetLSetupFrequency, nb::arg("cvode_mem"),
+      nb::arg("msbp"));
+
+m.def("CVodeSetMaxConvFails", CVodeSetMaxConvFails, nb::arg("cvode_mem"),
+      nb::arg("maxncf"));
+
+m.def("CVodeSetMaxErrTestFails", CVodeSetMaxErrTestFails, nb::arg("cvode_mem"),
+      nb::arg("maxnef"));
+
+m.def("CVodeSetMaxHnilWarns", CVodeSetMaxHnilWarns, nb::arg("cvode_mem"),
+      nb::arg("mxhnil"));
+
+m.def("CVodeSetMaxNonlinIters", CVodeSetMaxNonlinIters, nb::arg("cvode_mem"),
+      nb::arg("maxcor"));
+
+m.def("CVodeSetMaxNumSteps", CVodeSetMaxNumSteps, nb::arg("cvode_mem"),
+      nb::arg("mxsteps"));
+
+m.def("CVodeSetMaxOrd", CVodeSetMaxOrd, nb::arg("cvode_mem"), nb::arg("maxord"));
+
+m.def("CVodeSetMaxStep", CVodeSetMaxStep, nb::arg("cvode_mem"), nb::arg("hmax"));
+
+m.def("CVodeSetMinStep", CVodeSetMinStep, nb::arg("cvode_mem"), nb::arg("hmin"));
+
+m.def("CVodeSetMonitorFrequency", CVodeSetMonitorFrequency,
+      nb::arg("cvode_mem"), nb::arg("nst"));
+
+m.def("CVodeSetNonlinConvCoef", CVodeSetNonlinConvCoef, nb::arg("cvode_mem"),
+      nb::arg("nlscoef"));
+
+m.def("CVodeSetNonlinearSolver", CVodeSetNonlinearSolver, nb::arg("cvode_mem"),
+      nb::arg("NLS"));
+
+m.def("CVodeSetStabLimDet", CVodeSetStabLimDet, nb::arg("cvode_mem"),
+      nb::arg("stldet"));
+
+m.def("CVodeSetStopTime", CVodeSetStopTime, nb::arg("cvode_mem"),
+      nb::arg("tstop"));
+
+m.def("CVodeSetInterpolateStopTime", CVodeSetInterpolateStopTime,
+      nb::arg("cvode_mem"), nb::arg("interp"));
+
+m.def("CVodeClearStopTime", CVodeClearStopTime, nb::arg("cvode_mem"));
+
+m.def("CVodeSetEtaFixedStepBounds", CVodeSetEtaFixedStepBounds,
+      nb::arg("cvode_mem"), nb::arg("eta_min_fx"), nb::arg("eta_max_fx"));
+
+m.def("CVodeSetEtaMaxFirstStep", CVodeSetEtaMaxFirstStep, nb::arg("cvode_mem"),
+      nb::arg("eta_max_fs"));
+
+m.def("CVodeSetEtaMaxEarlyStep", CVodeSetEtaMaxEarlyStep, nb::arg("cvode_mem"),
+      nb::arg("eta_max_es"));
+
+m.def("CVodeSetNumStepsEtaMaxEarlyStep", CVodeSetNumStepsEtaMaxEarlyStep,
+      nb::arg("cvode_mem"), nb::arg("small_nst"));
+
+m.def("CVodeSetEtaMax", CVodeSetEtaMax, nb::arg("cvode_mem"),
+      nb::arg("eta_max_gs"));
+
+m.def("CVodeSetEtaMin", CVodeSetEtaMin, nb::arg("cvode_mem"), nb::arg("eta_min"));
+
+m.def("CVodeSetEtaMinErrFail", CVodeSetEtaMinErrFail, nb::arg("cvode_mem"),
+      nb::arg("eta_min_ef"));
+
+m.def("CVodeSetEtaMaxErrFail", CVodeSetEtaMaxErrFail, nb::arg("cvode_mem"),
+      nb::arg("eta_max_ef"));
+
+m.def("CVodeSetNumFailsEtaMaxErrFail", CVodeSetNumFailsEtaMaxErrFail,
+      nb::arg("cvode_mem"), nb::arg("small_nef"));
+
+m.def("CVodeSetEtaConvFail", CVodeSetEtaConvFail, nb::arg("cvode_mem"),
+      nb::arg("eta_cf"));
+
+m.def(
+  "CVodeSetRootDirection",
+  [](void* cvode_mem) -> std::tuple<int, int>
```

Comment:
Why is this returning a tuple? Guess we're missing a `_1d` suffix in the header file

Reply:
Yes, it is because we are missing the _1d suffix.